File: ExpenseManagerProject/Employee/menu.py
from expenses import (
    submit_expense,
    view_expenses,
    view_pending_expenses,
    view_expense_history,
    edit_expense,
    delete_expense
)

def employee_menu(user_id):
    while True:
        print("\n=== Employee Menu ===")
        print("1. Submit Expense")
        print("2. View All Expenses")
        print("3. View Pending Expenses")
        print("4. View Expense History")
        print("5. Edit Pending Expense")
        print("6. Delete Pending Expense")
        print("7. Logout")

        choice = input("Enter choice: ").strip()
        print()
        if choice == "1":
            submit_expense(user_id)
        elif choice == "2":
            view_expenses(user_id)
        elif choice == "3":
            view_pending_expenses(user_id)
        elif choice == "4":
            view_expense_history(user_id)
        elif choice == "5":
            edit_expense(user_id)
        elif choice == "6":
            delete_expense(user_id)
        elif choice == "7":
            print("Logging out...")
            break
        else:
            print("Invalid choice. Try again.")

File: ExpenseManagerProject/Manager/src/main/java/com/revature/manager/main/Menu.java
package com.revature.manager.main;

import com.revature.manager.service.ExpenseServiceImpl;
import com.revature.manager.service.ReportServiceImpl;

import java.util.InputMismatchException;
import java.util.List;
import java.util.Scanner;

public class Menu {
    private ExpenseServiceImpl expenseServiceImpl = new ExpenseServiceImpl();
    private ReportServiceImpl reportServiceImpl = new ReportServiceImpl();
    private Scanner sc  = new Scanner(System.in);

    public void managerMenu(int managerId){
        while(true){
            try {
                System.out.println("=== Manager Menu ===");
                System.out.println("1. View All Expenses");
                System.out.println("2. View Pending Expenses");
                System.out.println("3. Approve An Expense");
                System.out.println("4. Reject An Expense");
                System.out.println("5. Generate report");
                System.out.println("6. Logout");
                System.out.print("Choose an option: ");
                int choice = sc.nextInt();
                sc.nextLine();
                switch(choice){
                    case 1:
                        expenseServiceImpl.viewAllExpenses();
                        break;
                    case 2:
                        expenseServiceImpl.viewPendingExpenses();
                        break;
                    case 3:
                        expenseServiceImpl.approveService(managerId);
                        break;
                    case 4:
                        expenseServiceImpl.rejectService(managerId);
                        break;
                    case 5:
                        reportServiceImpl.generateReportMenu();
                        break;
                    case 6:
                        System.out.println("Logging out...");
                        return;

                    default:
                        System.out.println("Invalid choice. Please enter a number between 1 and 8.");
                        System.out.println();
                }
            }
            catch (InputMismatchException e){
                System.out.println("Invalid input. Please enter a valid number.");
                System.out.println();
                sc.nextLine();
            }
        }
    }
}
